### PR TITLE
Redirecting and global message

### PR DESCRIPTION
**Pull request by anunygard2**
+ mongoose version update

### CODE REVIEW

File: public/components/people/people.js [hunk 1]
```diff
@@ -2,42 +2,13 @@
...
-      'XML',
-      'PHP',
-      'JavaScript',
-      'CSS3'
-    ];
-
-    $scope.skills.sort();
-
+    if($rootScope.message) {
+      if($rootScope.message.section != 'people' || $rootScope.message.status == 'error' ){
```
Comment (lkduerig) [inline public/components/people/people.js]: Please use strict comparison in JS.

File: public/components/people/people.js [hunk 2]
```diff
@@ -59,41 +35,22 @@ angular.module('rmtPeopleController', [])
 
 // People form view
 
-  .controller('PeopleFormController', function($scope, $http, $stateParams, Persons, Person, Skills) {
+  .controller('PeopleFormController', function($rootScope, $scope, $http, $stateParams, $location, Persons, Person, Skills) {
     $scope.formData = {};
     $scope.id = $stateParams.id;
     $scope.showUpdateButtons = $scope.id.length>0 ? true : false;
```
Comment (lkduerig) [inline public/components/people/people.js]: Spaces missing here, e.g.: 
     $scope.showUpdateButtons = $scope.id.length > 0 ? true : false;
(I know this line isn't part of this commit, but it got noticed and now it is. I haven't even started on commenting syntax, yet. :)

File: public/components/people/people.js [hunk 3]
```diff
@@ -59,41 +35,22 @@ angular.module('rmtPeopleController', [])
...
-      'XML',
-      'PHP',
-      'JavaScript',
-      'CSS3'
-    ];
+    // get skills array
+    Skills.get()
+    .success(function(data) {
+        $scope.skills = data;
+
```
Comment (lkduerig) [inline public/components/people/people.js]: Remove blank line.

File: public/components/people/people.js [hunk 4]
```diff
@@ -59,41 +35,22 @@ angular.module('rmtPeopleController', [])
...
-      'Java',
-      'SQL',
-      'XML',
-      'PHP',
-      'JavaScript',
-      'CSS3'
-    ];
+    // get skills array
+    Skills.get()
+    .success(function(data) {
```
Comment (lkduerig) [inline public/components/people/people.js]: Missing a space, e.g. 
  .success(function (data) {
Also, indentation needs fixing for this block.

File: public/components/people/people.js [hunk 5]
```diff
@@ -59,41 +35,22 @@ angular.module('rmtPeopleController', [])
...
+        $scope.skills = data;
+
+    })
+    .error( function(){
```
Comment (lkduerig) [inline public/components/people/people.js]: Space before function should be removed, and two should be added around ():
  .error(function () {

File: public/components/people/people.js [hunk 6]
```diff
@@ -59,41 +35,22 @@ angular.module('rmtPeopleController', [])
...
+      console.log('error get')
```
Comment (lkduerig) [inline public/components/people/people.js]: Is this useful? (And should it have a semicolon at the end of the line?)

File: public/components/people/people.html [hunk 7]
```diff
@@ -1,9 +1,7 @@
-<section class="sections section-people clearfix submission--{{submissionSuccess}}">
+<section class="sections section-people clearfix submission--{{message.status}}">
  
   <div class="searchPeopleResult" id="SearchPeople" ng-controller="PeopleController">
-  	<div class="alert alert-{{submissionSuccess}}" role="alert" ng-show="submissionSuccess">{{submissionMessage}}</div>
-
-
+  	<div class="alert alert-{{message.status}}" role="alert" ng-if="message.msg">{{message.msg}}</div>
```
Comment (lkduerig) [inline public/components/people/people.html]: Nice. :+1: Good idea.

File: public/components/people/people.js [hunk 8]
```diff
@@ -142,13 +88,12 @@ angular.module('rmtPeopleController', [])
                 Persons.create($scope.formData)
                   // if creation is successful, call get function to get all the new stuff
                 .success(function(data) {
```
Comment (lkduerig) [inline public/components/people/people.js]: Space after function, and indentation.
  .success(function (data) {

File: public/components/people/people.js [hunk 9]
```diff
@@ -159,7 +104,6 @@ angular.module('rmtPeopleController', [])
             .success(function(data) {
```
Comment (lkduerig) [inline public/components/people/people.js]: space after function.

File: public/components/people/people.js [hunk 10]
```diff
@@ -173,12 +117,12 @@ angular.module('rmtPeopleController', [])
                 Person.put($scope.formData)
                 // if creation is successful, call get function to get all the new stuff
                 .success(function(data) {
```
Comment (lkduerig) [inline public/components/people/people.js]: Space after function and indentation.

File: public/components/people/people.js [hunk 11]
```diff
@@ -189,14 +133,15 @@ angular.module('rmtPeopleController', [])
             Persons.delete(id)
             // show remaining persons
             .success(function(data) {
```
Comment (lkduerig) [inline public/components/people/people.js]: Space after function and indentation.

File: public/components/people/people.js [hunk 12]
```diff
@@ -189,14 +133,15 @@ angular.module('rmtPeopleController', [])
...
-                $scope.showMessage('User ' +  name + ' deleted.', 'success');
+                $rootScope.message = { msg :'User ' + name + ' deleted.', section : 'people', status : 'success'};
+                $location.path('/api/people');
             });
         };
 
         $scope.addSkill = function(name){
```
Comment (lkduerig) [inline public/components/people/people.js]: Spaces around (name):
  function (name) {

File: public/components/projects/projects.js [hunk 13]
```diff
@@ -3,10 +3,13 @@
 angular.module('rmtProjectController', [])
     
     //Project list
-    .controller('ProjectController', function($scope, $http, Projects, Project, Persons, Person ) {
+    .controller('ProjectController', function( $rootScope, $scope, $http, Projects, Project, Persons, Person ) {
```
Comment (lkduerig) [inline public/components/projects/projects.js]: Review spacing.

File: public/components/projects/projects.js [hunk 14]
```diff
@@ -3,10 +3,13 @@
...
         $scope.formData = {};
-
- 
+        if($rootScope.message) {
```
Comment (lkduerig) [inline public/components/projects/projects.js]: Space after if, also in line below.

File: public/components/projects/projects.js [hunk 15]
```diff
@@ -3,10 +3,13 @@
...
+          if($rootScope.message.section != 'project' || $rootScope.message.status == 'error' ){
```
Comment (lkduerig) [inline public/components/projects/projects.js]: Please use strict comparison! And review spacing.

File: public/components/projects/projects.js [hunk 16]
```diff
@@ -24,11 +27,11 @@ angular.module('rmtProjectController', [])
     })
     
     //Project form
-    .controller('ProjectAddController', function($scope, $http, $stateParams, Projects, Project, Persons, Person ) {
+    .controller('ProjectAddController', function( $rootScope, $scope, $http, $stateParams, $location, Projects, Project, Persons, Person ) {
```
Comment (lkduerig) [inline public/components/projects/projects.js]: Review spacing.

File: public/components/projects/projects.js [hunk 17]
```diff
@@ -24,11 +27,11 @@ angular.module('rmtProjectController', [])
...
         $scope.formData = {};
-        $scope.projectParticipants = [];
         $scope.id = $stateParams.id;
         $scope.showUpdateButtons = $scope.id.length>0 ? true : false;
```
Comment (lkduerig) [inline public/components/projects/projects.js]: spaces around comparison operator.

File: public/components/projects/projects.js [hunk 18]
```diff
@@ -82,12 +67,12 @@ angular.module('rmtProjectController', [])
                 Projects.create($scope.formData)
                   // if creation is successful, call get function to get all the new stuff
```
Comment (lkduerig) [inline public/components/projects/projects.js]: Theoretically, comments should be correctly capitalized and punctuated. Some documentation parsers even depend on it. In the future, can we try to do that, and fix the ones that are not correct?

File: public/components/projects/projects.js [hunk 19]
```diff
@@ -119,12 +95,12 @@ angular.module('rmtProjectController', [])
                 Project.put($scope.formData)
                 // if creation is successful, call get function to get all the new stuff
                 .success(function(data) {
```
Comment (lkduerig) [inline public/components/projects/projects.js]: Spacing and indentation.

File: public/components/people/people.html (hunk 7 above)
Comment (amolnw2778) [inline public/components/people/people.html]: Please use two spaces indentation always for all files.

File: public/components/projects/projects.js [hunk 20]
```diff
@@ -134,59 +110,28 @@ angular.module('rmtProjectController', [])
             Projects.delete(id)
             // show remaining persons
             .success(function(data) {
```
Comment (lkduerig) [inline public/components/projects/projects.js]: Spacing and indentation.

File: public/components/projects/projects.js [hunk 21]
```diff
@@ -134,59 +110,28 @@ angular.module('rmtProjectController', [])
...
-          }
-        }
-        else {
-          return true;
-        }
-      };
-   
   })
-.directive('projectPersons',[
+.directive('projectsForm',[
```
Comment (lkduerig) [inline public/components/projects/projects.js]: Spacing.

File: public/components/people/people.js [hunk 22]
```diff
@@ -2,42 +2,13 @@
...
-      'SQL',
-      'XML',
-      'PHP',
-      'JavaScript',
-      'CSS3'
-    ];
-
-    $scope.skills.sort();
-
+    if($rootScope.message) {
```
Comment (amolnw2778) [inline public/components/people/people.js]: Add space after `if`

File: public/components/projects/projects.js [hunk 23]
```diff
@@ -134,59 +110,28 @@ angular.module('rmtProjectController', [])
...
+    function(){
```
Comment (lkduerig) [inline public/components/projects/projects.js]: spacing.
  function () {

File: public/components/people/people.js (hunk 1 above)
Comment (amolnw2778) [inline public/components/people/people.js]: Add space after `if`, before closing brace `{`. Remove extra spacing after `error`

File: public/components/people/people.html (hunk 7 above)
Comment (lkduerig) [inline public/components/people/people.html]: I am not sure about this. I prefer 2 spaces, but that is more of a Drupal standard, Javascript often recommends 4. But we should agree on one or the other and be consistent.

File: public/components/people/people.js (hunk 1 above)
Comment (amolnw2778) [inline public/components/people/people.js]: Use strict check here onward please.

File: public/components/people/people.js [hunk 24]
```diff
@@ -2,42 +2,13 @@
...
+      if($rootScope.message.section != 'people' || $rootScope.message.status == 'error' ){
+        $rootScope.message = {}
```
Comment (amolnw2778) [inline public/components/people/people.js]: Missing `;` at the end. Not mandatory, but a good practice.

File: public/components/people/people.js [hunk 25]
```diff
@@ -45,9 +16,14 @@ angular.module('rmtPeopleController', [])
     
     Persons.get()
     .success(function(data) {
```
Comment (amolnw2778) [inline public/components/people/people.js]: Required 2 spaces indentation here.

File: public/components/people/people.js [hunk 26]
```diff
@@ -45,9 +16,14 @@ angular.module('rmtPeopleController', [])
...
-        //$scope.skills = skills;
         $scope.persons = data;
-
+        
+        // Person skills to string mode for filtering
+        _.each ($scope.persons, function(person) {
+          if(!angular.isUndefined(person.skills)){
```
Comment (amolnw2778) [inline public/components/people/people.js]: Missing spaces - after if, before {

File: public/components/people/people.js [hunk 27]
```diff
@@ -45,9 +16,14 @@ angular.module('rmtPeopleController', [])
...
+            person.skillsList = person.skills.toString();
+          }
+        });
     })
     .error( function(){
       console.log('error get')
```
Comment (amolnw2778) [inline public/components/people/people.js]: Better to make error messages public instead showing it in console.

File: public/components/people/people.js (hunk 2 above)
Comment (amolnw2778) [inline public/components/people/people.js]: Better `$scope.showUpdateButtons = $scope.id.length > 0;`

File: public/components/people/people.js [hunk 28]
```diff
@@ -59,41 +35,22 @@ angular.module('rmtPeopleController', [])
 
 // People form view
 
-  .controller('PeopleFormController', function($scope, $http, $stateParams, Persons, Person, Skills) {
+  .controller('PeopleFormController', function($rootScope, $scope, $http, $stateParams, $location, Persons, Person, Skills) {
     $scope.formData = {};
     $scope.id = $stateParams.id;
     $scope.showUpdateButtons = $scope.id.length>0 ? true : false;
+    $scope.skills = [];
+    $rootScope.message = {}
```
Comment (amolnw2778) [inline public/components/people/people.js]: Missing `;` at the end.

File: public/components/people/people.js [hunk 29]
```diff
@@ -59,41 +35,22 @@ angular.module('rmtPeopleController', [])
...
-      'Documentation / Specifier',
-      'Database Expert',
-      'Java',
-      'SQL',
-      'XML',
-      'PHP',
-      'JavaScript',
-      'CSS3'
-    ];
+    // get skills array
```
Comment (amolnw2778) [inline public/components/people/people.js]: First letter should be uppercase. Also missing `.` at the end.

File: public/components/people/people.js (hunk 6 above)
Comment (amolnw2778) [inline public/components/people/people.js]: See my previous comment about messages.

File: public/components/people/people.js [hunk 30]
```diff
@@ -142,13 +88,12 @@ angular.module('rmtPeopleController', [])
                 Persons.create($scope.formData)
                   // if creation is successful, call get function to get all the new stuff
                 .success(function(data) {
-                    var personName = $scope.formData.name;
-                    $scope.showMessage('User ' + personName + ' added.', 'success');
+                    $rootScope.message = { msg :'User ' + $scope.formData.name + ' added.', section : 'people', status : 'success'};
+                    $location.path('/api/people');
                 });
             }
```
Comment (amolnw2778) [inline public/components/people/people.js]: Proper indentation required.

File: public/components/people/people.js [hunk 31]
```diff
@@ -173,12 +117,12 @@ angular.module('rmtPeopleController', [])
...
                 .success(function(data) {
-                    var personName = $scope.formData.name;
-                    $scope.showMessage('User ' + personName + ' updated');
+                    $rootScope.message = { msg :'User ' + $scope.formData.name + ' updated.', section : 'people', status : 'success'};
+                    $location.path('/api/people');
                 });
             }
             else {
-              $scope.showMessage('Please fill all required fields.', 'error');
+              $rootScope.message = { msg :'Please fill all required fields.', section : 'people', status : 'error'};
```
Comment (amolnw2778) [inline public/components/people/people.js]: Wrong spacings

File: public/components/people/peopleForm.html [hunk 32]
```diff
@@ -1,5 +1,5 @@
 <div id="AddPeople" ng-controller="PeopleFormController">
```
Comment (amolnw2778) [inline public/components/people/peopleForm.html]: Need to have uniformity in ID, class naming conventions. ie. lower/upper/camel cases.

File: public/components/projects/projects.js (hunk 17 above)
Comment (amolnw2778) [inline public/components/projects/projects.js]: Better `$scope.showUpdateButtons = $scope.id.length > 0;`

File: public/components/projects/projectsForm.html [hunk 33]
```diff
@@ -1,9 +1,6 @@
-<section class="sections section-people submission--{{submissionSuccess}} clearfix"
+<section class="sections section-people submission--{{message.status}} clearfix"
          ng-controller="ProjectAddController">
```
Comment (amolnw2778) [inline public/components/projects/projectsForm.html]: This can be in single line.

File: public/components/projects/projectsList.html [hunk 34]
```diff
@@ -0,0 +1,39 @@
+<section class="sections section-people clearfix submission--{{message.status}}" ng-controller="ProjectController">
+  <div class="alert alert-{{message.status}}" role="alert" ng-if="message.msg">{{message.msg}}</div>
+      <div class="searchProjectResult filter filter--projects" id="SearchProject">
+
+        <div class="filter--controls">
+          <input type="text" class="form-control" placeholder="Search project" ng-model="searchProject" />
```
Comment (amolnw2778) [inline public/components/projects/projectsList.html]: Closing `/` not required for input/img/br/hr fields in HTML5

File: public/routes.js [hunk 35]
```diff
@@ -1,108 +1,3 @@
...
-        res.sendfile('./index.html'); // load the single view file (angular will handle the page changes on the front-end)
-    });
-
-}
-
-
-*/
-
-
 var Person      = require('./models/person');
```
Comment (amolnw2778) [inline public/routes.js]: Too much spacing between variable and `=`

File: public/routes.js [hunk 36]
```diff
@@ -291,18 +186,38 @@ module.exports = function(router) {
...
-                // remove .png
-                var imageName = list[i].substring(0, list[i].indexOf('.') );
-                resp.push({name : imageName});
-              }
-              res.json(resp);
-            });
-
+        
+            var skills = [
+              'Notepad',
```
Comment (amolnw2778) [inline public/routes.js]: `Notepad, Notepad++` are skills? Please put real skills only.

**Pull request merged**